### PR TITLE
docs: Controlled Select with `allowClear`

### PR DESCRIPTION
**Pull request by MoyuScript**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 💡 Background and solution

业务中有使用到需要清除的受控下拉组件，目前文档对于这种情况还不是很完善，所以帮忙完善一下。

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   添加了一些下拉框示例和完善 API 文档        |
| 🇨🇳 Chinese | Added some demo for select and completed API docs   |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

**Comment by MadCcc** [comment]
跑一下 `npm run test`，会有些 snapshot 的更新。

File: components/select/demo/controlled-with-allowClear.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+---
+order: 1
+title:
+  zh-CN: 可清除的受控下拉组件
+  en-US: Controlled Select with `allowClear`
+---
+
+## zh-CN
+
+可清除的受控下拉组件
```
Comment (afc163) [inline components/select/demo/controlled-with-allowClear.md:10]:
```suggestion
可清除的受控下拉组件。
```

File: components/select/demo/controlled-with-allowClear.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  const [value, setValue] = useState(null);
+  const [loading, setLoading] = useState(false);
+
+  function onChange(newValue) {
+    setLoading(true);
+    // The `newValue` should be `undefined` when user cleared the select.
+    // Delay 1 seconds then update the value.
+    setTimeout(() => {
+      setLoading(false);
+      setValue(newValue === undefined ? null : newValue);
```
Comment (afc163) [inline components/select/demo/controlled-with-allowClear.md:32]: 这是 bug 吧，默认应该 onChange 里无脑 setValue(value) 就应该生效了。

cc @MadCcc

Comment (MoyuScript) [reply components/select/demo/controlled-with-allowClear.md:32]: value 为 undefined 的时候会导致组件不受控，可以看看这个：<https://codesandbox.io/s/dai-sou-suo-kuang-antd-4-19-0-forked-bkvzbx?file=/index.js>

Comment (afc163) [reply components/select/demo/controlled-with-allowClear.md:32]: 这个要看一下，我们不能推荐用户在 onChange 这么写代码。

Comment (afc163) [reply components/select/demo/controlled-with-allowClear.md:32]: cc @zombieJ

**Comment by afc163** [comment]
trace #34437

**Comment by guiyu1230** [comment]
props有无value属性 和  props有value熟悉但值为undefined 是不一样的. 我的观点是
`'value' in props ? props.value : state.value`

**Pull request closed without merging**